pitch.md

```diff
@@ -23,38 +35,187 @@ should be more specific than "The user describes properties of unicorns.", e.g.,
+play could be to annotate when a player takes a long time to think or when a
+player plays a card very quickly, as sometimes this might influence how one
+decides to play the hand.  The auction is a bit more complex.  At a basic level
+players must be able to specify what bids were made and the order the bids were
+made in.  However, in bridge people play different "systems" which assign artificial
+meanings to the bids. There must be a way to communicate this information as well.
+I would propose two methods for communicating this information.  The user can have
+the ability to describe the systems played by both partnerships (bridge is a 4
+player game where there are 2 partnerships playing against eachother).  In addition
+or instead of this there should also be the ability to annotate particular bids
+that have some unusual meaning.  The advantage of this two pronged approach is
+that describing a system can be very succinct an easy. For instance the user might
+just be fine saying the one partnership is playing "2/1" and the other partnership
+is playing "precision" and not defining any of the particular bids but letting the
+reader look up those systems to find the meanings of the bids.  But in the case
+that people are playing a very strange system or a system that the person writing
+about the hand doesn't think the readers will know about, they can go in and give
+explanations of any unusual bids. A third type of information that could be useful
+would be general information about thoughts going into the hand.  For example
+players might play differently if they think they are way behind in a team game
+than if they think they have a comfortable lead.
+
+I'm currently thinking that the user will input the hand in a format that is
+convenient to be inputted for them and then the program will handle displaying
+the data in an easy to read manner for people wanting to analyze the hand. I think
+a hand diagram in text format might make sense for the output. Something like
+```
+        xx
+        xx
+        xxxxx
+        xxxx
+xxxx           xxx
+xxxx           xxxx
+xxxx           x
+x              xxxxx
+
+        xxxx
+        xxx
+        xxx
+        xxx
+```
+where x's are replaced with the actual cards.
 
 ## Operation (semantics)
 _What might happen when a program runs? Is there a computational model that
 corresponds with your domain? How does a program interact with the
 user? What kinds of semantic (i.e., non-syntax) errors might occur, and how 
 might they be communicated to the user?_
 
+When a program runs it will take the input by the user and display it in a nice
+manner.  This might consist of creating a bridge diagram based off the input. It
+could also be showing some end position -- that is showing the state of the cards
+after some number of cards have been played. Another useful thing a user could do
+when running the program would be to show some of the hands but not others. This
```

Comment:
So would the users have to know all the hands of all the other players? Would there be a way so the user could let the computer do that? In other words, could the user say that they know their hand and their partner's hand but not their opponents' hands and have the computer generate legal hands for their opponents?